src/firmware/shared/circular_buffer.h

```diff
@@ -0,0 +1,41 @@
+#include <math.h>
+#include <stdbool.h>
+#include <stdlib.h>
+
+#include "shared/constants.h"
+
+struct circular_buffer
+{
+    float *bufferData;
+    int head;
+    int tail;
+    bool isFull;
+};
+
+/**
+ * Initialize the circular_buffer struct members
+ * @param  cbuff The circular_buffer struct
+ */
+void circular_buffer_init(struct circular_buffer *cbuff);
+
+/**
+ * Free the circular_buffer
+ * @param  cbuff The circular_buffer struct
+ */
+void circular_buffer_free(struct circular_buffer *cbuff);
```

Comment:
`circular_buffer_free -> circular_buffer_destroy`, take a look at `firmware_world.h` for the example function signatures

Reply:
Done.